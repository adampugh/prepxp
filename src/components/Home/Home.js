import React from "react";

import Navbar from "../UI/navbarLoggedOut";
import HomeBanner from "./homeBanner";
import HomeAbout from "./homeAbout";
import HomePanels from "./homePanels";
import HomeCarousel from "./homeCarousel";
import HomeBottomBanner from "./homeBottomBanner";
import Footer from "../UI/footer";


const Home = () => (
    <div>
        <Navbar />
        <HomeBanner />
        <HomeAbout />
        <HomePanels />
        <HomeCarousel />
        <HomeBottomBanner />
        <Footer />
    </div>
);

export default Home;