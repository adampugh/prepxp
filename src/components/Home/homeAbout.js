import React from "react";
import { Link } from "react-router-dom";

import interviewImg from "../../assets/images/interview.png";

const homeAbout = () => (
    <main>
        <div className="about__title">
            <h1>Why use PrepXP in your job search?</h1>
        </div>
        <div className="about__grid">
            <div>
                <img className="bannerText__image" src={interviewImg} alt="arm icon" />
            </div>
            <div className="about__grid__text">
                <h1>Need help with upcoming interviews?</h1>
                <p>
                    PrepXP aids users in landing their dream jobs by helping them pass the interview.
                    Checkout our curated lists based on questions asked by some of the
                    largest companies in the world.
                </p>
                <Link to="/prexp/login"><button className="btn">Start now</button></Link>
            </div>
        </div>
    </main>
);

export default homeAbout;