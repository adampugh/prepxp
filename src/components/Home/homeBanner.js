import React from "react";

import desk from "../../assets/images/deskimg.png";

const homeBanner = () => (
    <section className="homeBanner">
        <div className="container">
            <div className="homeBanner__content">
                
                <div className="homeBanner__content__textPanel">
                    <h1>
                        Prepare for success
                    </h1>
                    <p>Prepare for job interviews by answering pracitcal interview questions!
                    Create and build your own lists of interview questions or import from curated lists.</p>
                    <button className="btn">Get Started</button>
                </div>
                <div className="homeBanner__content__picturePanel">
                    <img className="bannerText__image" src={desk} alt="arm icon" />
                </div>
            </div>
        </div>
    </section>
);

export default homeBanner;



// <section className="homeBanner">
// <div className="container">
//     <div className="homeBanner__content">
//         <h1>
//             Prepare for job interviews by answering pracitcal interview questions!
//         </h1>
//         <button className="btn">Sign Up</button>
//     </div>
// </div>
// </section>