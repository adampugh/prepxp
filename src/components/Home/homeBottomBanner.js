import React from "react";
import { Link } from "react-router-dom";

import bottomBanner from "../../assets/images/bottomBanner.png";

const homeBottomBanner = () => (
    <section className="bottomBanner">
        <div className="container">
            <h1>Land your dream job</h1>
            <p>Sign up to PrepXP today to better your career and get the job 
            you deserve! Pass the interview by gaining experience and preparing
            for success in all future job interviews.</p>
            <Link to="/login"><button className="btn">Start now</button></Link>
            <img src={bottomBanner} alt="policeman, business man and designer" />
        </div>
    </section>
);

export default homeBottomBanner;
