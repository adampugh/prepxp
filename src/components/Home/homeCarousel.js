import React from "react";

import carousel1 from "../../assets/images/carousel1.jpg";
import carousel2 from "../../assets/images/carousel2.jpg";
import carousel3 from "../../assets/images/carousel4.jpg";

const Carousel = require("nuka-carousel");



const homeCarousel = () => (
    <section className="container carousel">
        <Carousel autoplay={true} wrapAround={true} >
            <div className="carousel__panel">
                <img src={carousel3} alt="man climbing chart" />
                <div>
                    <h2>Available on all devices</h2>
                    <p>Whether you are accessing PrepXP on a mobile, tablet or desktop
                    rest assured that you are fully covered. PrepXP is a fully responsive
                    web application that allows users to have full functionality on all
                    popular browsers.</p>
                    <p>Sign up today to hone your skills and be prepared whatever devices
                    you are using</p>
                </div>
            </div>
            <div className="carousel__panel">
                <img src={carousel1} alt="man climbing chart" />
                <div>
                    <h2>Practise anywhere</h2>
                    <p>Practise anywhere you have an internet connection. PrepXP is
                    perfect for last minute preparations before a job interview or
                    while you're preparing to re-enter the workforce.
                    </p>
                    <p>Have the freedom to travel and know that your skills are still
                    being honed wherever you are in the world. PrepXP will be a 
                    available as an offline web application in the future.</p>
                </div>
            </div>
            <div className="carousel__panel">
                <img src={carousel2} alt="man climbing chart" />
                <div>
                    <h2>Save valuable time</h2>
                    <p>Finding quality up to date interview questions are prepartion can 
                    be costly and difficult to find. At PrepXP we value your time and
                    want to give the best service possible.</p>
                    <p>We curate lists of interview questions that are up to date and 
                    relevant to the current job market to ensure that you are well 
                    equipped to pass any interview</p>
                </div>
            </div>
        </Carousel>
    </section>
);

export default homeCarousel;