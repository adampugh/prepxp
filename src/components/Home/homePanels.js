import React from "react";

import panel1Img from "../../assets/images/panel1.jpg";
import panel2Img from "../../assets/images/panel2.jpg";
import panel3Img from "../../assets/images/panel3.jpg";

const homePanels = () => (
    <section className="container">
        <div className="panels__grid">
            <div>
                <img src={panel1Img} alt="man climbing chart" />
                
                <h2>Develop skills</h2>
                <p>Practise your ability to answer difficult interview questions and 
                    learn how to pass the interview.</p>
                
            </div>
            <div>
                <img className="bannerText__image" src={panel2Img} alt="arm icon" />
                <h2>Beat competition</h2>
                <p>Gain insight to some of the most common interview questions and 
                    prepare excellent answers to help you stand out.</p>
            </div>
            <div>
                <img className="bannerText__image" src={panel3Img} alt="arm icon" />
                <h2>Feel confident</h2>
                <p>Prepare fantastic answers and run through mock interviews to gain 
                    experience so that you feel relaxed in any interview.</p>
            </div>
        </div>
    </section>
);

export default homePanels;