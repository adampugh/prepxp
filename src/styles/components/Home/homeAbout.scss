main {
    text-align: center;
    background: $purple-gradient, url("../assets/images/bg3.jpg") no-repeat center center; 
    background-size: cover; 
    padding-bottom: $padding-m;
    .about__title {
        color: white;
        padding: $padding-m 0;
        h1 {
            @include header;
            width: 70%;
            margin: 0 auto;
        }
    }
}

.about__grid {
    display: grid;
    grid-template-columns: 1fr;
    width: 70%;
    margin: 0 auto;
    text-align: center;
    h1 {
        @include header;
        color: $off-grey;
        font-size: $font-l;
    }
    p {
        font-size: $font-s;
        color: $light-grey;
    }
    img {
        width: 90%;
        padding-bottom: $padding-s;
    }
}

.about__grid__text {
    text-align: left;
    padding: 30px 30px;
    margin: 0 auto;
    background: white;
    border-radius: 15px;
    @include boxshadow;
}



@media (min-width: $desktop-breakpoint) {
    main {
        padding-bottom: 0
    }
    .about__grid {
        grid-template-columns: 2fr 1fr;
        text-align: left;
        padding: $padding-m 0;
        img {
            padding: 0;
        }
    }
}