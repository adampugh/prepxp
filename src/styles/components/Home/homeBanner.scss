.homeBanner {
    padding: $padding-l 0 $padding-m 0;
    // text-align: center;
    // font-size: $font-m;
    // color: white;
    background: $black-gradient, url("../assets/images/bg5.jpg") no-repeat center; 
    background-size: cover; 
    h1 {
        // padding-bottom: 20px;
        // text-shadow: 1px 2px 5px rgba(14, 14, 14, 0.241);
    }
    img {
        width: 100%;
        max-width: 800px;
    }
}

.homeBanner__content__textPanel {
    text-align: left;
    padding: 30px 30px;
    width: 70%;
    margin: 0 auto;
    h1 {
        @include header;
        color: white;
    }
    p {
        margin: 0;
        padding: 20px 0 50px 0;
        font-size: $font-s;
        color: rgb(157, 157, 157);
    }
}

.btn {
    border-radius: 55px;
    background: white;
    color: $off-purple;
    padding: 15px 40px;
    font-size: .8em;
    font-weight: 600;
    border: $off-purple 2px solid;
    text-transform: uppercase;
    cursor: pointer;
    @include boxshadow;
}

@media (min-width: $desktop-breakpoint) {
    .btn {
        -webkit-transition: background-color .5s ease-out;
        -moz-transition: background-color .5s ease-out;
        -o-transition: background-color .5s ease-out;
        transition: background-color .5s ease-out;
        &:hover {
            background-color: $off-purple;
            border: white solid 2px;
            color: white;
        }
    }
}




@media (min-width: $desktop-breakpoint) {
    .homeBanner__content {
        display: grid;
        grid-template-columns: 1fr 1fr;
    }

    .homeBanner__content__textPanel {
        margin: 0;
        background: white;
        border-radius: 15px;
        @include boxshadow;
        h1 {
            color: $off-grey;
        }
    }

    .homeBanner__content__picturePanel {
        align-self: end;
        img {
            position: relative;
            bottom: -15px;
        }
    }
}

