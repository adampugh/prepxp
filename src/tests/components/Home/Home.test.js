import React from "react";
import { shallow } from "enzyme";

import Home from "../../../components/Home/Home";

describe("<Home />", () => {
    it("should match snapshots", () => {
        const wrapper = shallow(<Home />);
        expect(wrapper).toMatchSnapshot();
    });
})