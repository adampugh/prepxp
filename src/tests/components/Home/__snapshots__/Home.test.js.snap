// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshots 1`] = `
<div>
  <Navbar />
  <homeBanner />
  <homeAbout />
  <homePanels />
  <homeCarousel />
  <homeBottomBanner />
  <footer />
</div>
`;
